packages/@aws-cdk/aws-codedeploy/test/server/test.deployment-group.ts

```diff
@@ -28,6 +28,19 @@ export = {
       test.done();
     },
 
+    'creating an application with physical name if needed'(test: Test) {
+      const stack = new cdk.Stack();
+      new codedeploy.ServerDeploymentGroup(stack, 'MyDG', {
+        deploymentGroupName: cdk.PhysicalName.GENERATE_IF_NEEDED,
+      });
+
+      expect(stack).to(haveResource('AWS::CodeDeploy::Application', {
+        'ComputePlatform': 'Server',
```

Comment:
We do need an `ApplicationName` provided here.

If it doesn't appear, then the fix is not sufficient. It may be the that the ServerApplication does not respect PhysicalNames, for example. 

Try and step through the code to see where the discrepancy between what we expect and what actually happens comes from.

Reply:
Can we give a custom name and check whether we'll be receiving the out as expected instead of a PhysicaName?